nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/main/java/org/apache/nifi/redis/state/RedisStateProvider.java

```diff
@@ -106,6 +122,16 @@ public final void initialize(final StateProviderInitializationContext context) t
                             + " is configured in clustered mode, and this service requires a non-clustered Redis")
                     .build());
         }
+        final boolean enableTls = validationContext.getProperty(ENABLE_TLS).asBoolean();
+        if (enableTls && sslContext == null) {
+            results.add(new ValidationResult.Builder()
+                    .subject(ENABLE_TLS.getDisplayName())
+                    .valid(false)
+                    .explanation(ENABLE_TLS.getDisplayName()
+                            + " is set to 'true', but Apache NiFi is not secured.  This state provider can only use a TLS-enabled connection " +
+                            "if a keystore and truststore is provided in nifi.properties.")
```

Comment:
Minor grammatical adjustment:
```suggestion
                            "if a keystore and truststore are provided in nifi.properties.")
```